packages/databricks-vscode/src/ui/bundle-resource-explorer/JobTreeNode.ts

```diff
@@ -0,0 +1,115 @@
+import {TreeItemCollapsibleState, ExtensionContext} from "vscode";
+import {BundleRemoteState} from "../../bundle/models/BundleRemoteStateModel";
+import {
+    BundleResourceExplorerResource,
+    BundleResourceExplorerTreeItem,
+    BundleResourceExplorerTreeNode,
+} from "./types";
+import {BundleRunStatusManager} from "../../bundle/run/BundleRunStatusManager";
+import {ContextUtils} from "./utils";
+import {ConnectionManager} from "../../configuration/ConnectionManager";
+import {JobRunStatusTreeNode} from "./JobRunStatusTreeNode";
+import {JobRunStatus} from "../../bundle/run/JobRunStatus";
+import {TaskTreeNode} from "./TaskTreeNode";
+import {TaskHeaderTreeNode} from "./TaskHeaderTreeNode";
+
+export class JobTreeNode implements BundleResourceExplorerTreeNode {
+    readonly type = "jobs";
+    get url(): string | undefined {
+        const host = this.connectionManager.databricksWorkspace?.host;
+
+        if (host === undefined) {
+            return undefined;
+        }
+
+        return `${host.toString()}jobs/${this.data.id}`;
+    }
+
+    constructor(
+        private readonly context: ExtensionContext,
+        private readonly bundleRunStatusManager: BundleRunStatusManager,
+        private readonly connectionManager: ConnectionManager,
+        public readonly resourceKey: string,
+        public readonly data: BundleResourceExplorerResource<"jobs">,
+        public parent?: BundleResourceExplorerTreeNode
+    ) {}
+
+    isRunning(resourceKey: string) {
+        const runner = this.bundleRunStatusManager.runStatuses.get(resourceKey);
+        return runner?.runState === "running";
+    }
+
+    getTreeItem(): BundleResourceExplorerTreeItem {
+        const isRunning = this.isRunning(this.resourceKey);
+
+        return {
+            label: this.data.name,
+            contextValue: ContextUtils.getContextString({
+                resourceType: this.type,
+                running: isRunning,
+                hasUrl: this.url !== undefined,
+                nodeType: this.type,
+            }),
+            collapsibleState: isRunning
+                ? TreeItemCollapsibleState.Collapsed
+                : TreeItemCollapsibleState.Expanded,
+        };
+    }
+
+    getChildren(): BundleResourceExplorerTreeNode[] {
+        if (this.data.tasks === undefined) {
+            return [];
+        }
+
+        const children: BundleResourceExplorerTreeNode[] = [];
+        const runMonitor = this.bundleRunStatusManager.runStatuses.get(
+            this.resourceKey
+        ) as JobRunStatus | undefined;
+
+        if (runMonitor?.runId !== undefined) {
+            children.push(new JobRunStatusTreeNode(runMonitor));
```

Comment:
I guess we should add `this` as a "parent" argument to the JobRunStatusTreeNode constructor.
We are using the `parent` constructor argument inconsistently, might be better to change the constructor types of all the elements that should have parents to have non optional `parent` argument.

On the other hand we can remove all the "parent" handling, since we aren't using the reveal API and don't have to implement `getParent` method in our tree provider.

Reply:
>On the other hand we can remove all the "parent" handling, since we aren't using the reveal API and don't have to implement getParent method in our tree provider.

I want to use it in the future to jump to say a pipeline from a pipeline task etc.